cmd/networking/istio/revision.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/pkg/apis"
+	"knative.dev/serving/pkg/apis/serving"
+)
+
+// RevisionStub is stub for the versioned types for the purposes
+// of defaulting
+type RevisionStub struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// TODO - add a general solution that captures
+	// top level properties in addition to spec & status
+	Spec   runtime.RawExtension `json:"spec"`
+	Status runtime.RawExtension `json:"status"`
+}
+
+// SetDefaults will apply istio specific defaults to the RevisionStub
+func (r *RevisionStub) SetDefaults(context.Context) {
+	if r.Labels == nil {
+		r.Labels = make(map[string]string)
+	}
+
+	service, ok := r.Labels[serving.ServiceLabelKey]
+
+	// Shouldn't happen but you never know
```

Comment:
Why shouldn't this happen? The label is only set for revisions that have been generated via a service -> configuration.

Reply:
Good point using the configuration label would make more sense

Reply:
I attempt to use both now